datacite-draft-doi-handler/src/test/java/no/unit/nva/doi/DraftDoiHandlerTest.java

```diff
@@ -0,0 +1,110 @@
+        handler = new DraftDoiHandler();
+        outputStream = new ByteArrayOutputStream();
+        context = mock(Context.class);
+        event = mock(AwsEventBridgeEvent.class);
+    }
+
+
+    @Test
+    public void handleRequestReturnsDoiUpdateDtoWithPublicationUriWhenInputIsValid() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_valid.json"));
+        handler.handleRequest(inputStream,outputStream,context);
+        DoiUpdateDto response = parseResponse();
+        assertThat(response.getPublicationId(), is(not(nullValue())));
+    }
+
+    private DoiUpdateDto parseResponse() {
+        return attempt(() -> JsonUtils.objectMapper.readValue(outputStream.toString(), DoiUpdateDto.class))
+            .orElseThrow();
+    }
+
+    @Test
+    public void handleRequestThrowsExceptionOnMissingEventItem() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_empty_item.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> handler.handleRequest(inputStream, outputStream, context));
+
+        MatcherAssert.assertThat(exception.getMessage(), is(DraftDoiHandler.PUBLICATION_IS_MISSING_ERROR));
+    }
+
+    @Test
+    public void handleRequestThrowsExceptionOnMissingCustomerId() {
+        InputStream inputStream = IoUtils.inputStreamFromResources(
+            Path.of("doi_publication_event_empty_institution_owner.json"));
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+            () -> handler.handleRequest(inputStream, outputStream, context));
+
+        MatcherAssert.assertThat(exception.getMessage(), is(DraftDoiHandler.CUSTOMER_ID_IS_MISSING_ERROR));
+    }
+
+    @Test
+    public void draftDoiHandlerReturnsDoiUpdateDtoOnValidPublicationDto() {
+        Publication publication = new PublicationDtoTestDataGenerator().createRandomStreamRecord().asPublicationDto();
+        PublicationHolder publicationHolder = new PublicationHolder(DOI_PUBLICATION, publication);
+
+        DoiUpdateDto doiUpdateDto = handler.processInputPayload(publicationHolder, event, context);
+
+        assertThat(doiUpdateDto, doesNotHaveNullOrEmptyFields());
+    }
+
+    @Test
+    public void draftDoiHandlerThrowsExceptionOnInvalidPublicationDto() {
+        PublicationHolder publicationHolder = getFailingPublicationHolder();
+
+        RuntimeException exception = assertThrows(
+            RuntimeException.class,
+            () -> handler.processInputPayload(publicationHolder, event, context)
+        );
+
+        assertThat(exception.getMessage(), equalTo(ERROR_MESSAGE));
+    }
+
+    private PublicationHolder getFailingPublicationHolder() {
+        PublicationHolder publicationHolder = mock(PublicationHolder.class);
+        when(publicationHolder.getItem()).thenThrow(new RuntimeException(ERROR_MESSAGE));
```

Comment:
when will PublicationHolder throw a runtime exception? :o

Reply:
Changed to IllegalArgumentException.